src/stores/ledger.ts

```diff
@@ -339,17 +341,14 @@ class LedgerHub implements LedgerTaskRunner {
             account = await comm.getPubKey(accountPath) as LedgerAccount;
           }
 
-          unsignedTx.senderPublicKey = account.publicKey;
-          const txBytes = unsignedTx.getBytes(true, true);
+          unsignedTx.senderPublicKey = Buffer.from(account.publicKey, 'hex') as Buffer & As<'publicKey'>;
+          const txBytes = Rise.txs.bytes(unsignedTx);
 
-          let signedTx: null | ITransaction;
+          let signedTx: null | PostableRiseTransaction;
           transport.setExchangeTimeout(30000);
           try {
-            unsignedTx.signature = await comm.signTX(accountPath, txBytes, false);
-            signedTx = {
-              ...unsignedTx.toObj(),
-              senderId: account.address,
-            };
+            unsignedTx.signature = await comm.signTX(accountPath, txBytes, false) as Buffer & As<'signature'>;
+            signedTx = Rise.txs.toPostable(unsignedTx);
```

Comment:
The _senderId_ field isn't needed anymore?

Reply:
https://github.com/vekexasia/dpos-offline/blob/master/src/codecs/rise.ts#L26

Reply:
And the resolve closure could be simplified as well then:

```
       const task = new LedgerTask(
         channelId,
         ({ account, signedTx }) => {
           // Add the account to the cache if we had cache-miss in the task runner
           if (!this.accountCache[accountSlot]) {
             this.accountCache[accountSlot] = account;
           }
           resolve(signedTx);
         },
         reject,
```

would become

```
       const task = new LedgerTask(
         channelId,
         resolve,
         reject,
```